src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1603,6 +1607,12 @@ ss::future<add_paritions_tx_reply> tx_gateway_frontend::do_add_partition_to_tx(
                 res_partition.partition_index = req_partition;
                 res_partition.error_code = tx_errc::none;
                 res_topic.results.push_back(res_partition);
+            } else if (
+              disabled_set && disabled_set->is_disabled(req_partition)) {
```

Comment:
nit: maybe we can add a method `bool is_disabled() const` in `metadata_cache` as i see this patter repeating all over the PR